### ISSUE TITLE
Batch processing/batch normalization for SOM

### ISSUE THREAD

**Issue by ngreenwald** [angelolab/ark-analysis#384]
**Is your feature request related to a problem? Please describe.**
The current SOM code iterates through each point one at a time. I think there might be some speedups if we can batch this for multiple points. However, more importantly, I think we're less likely to run into problems with the normalization if we have data from multiple FOVs. 99.9% normalization from a single FOV could lead to weird results. 

**Describe the solution you'd like**
An option to specify a bath size, both for training and running, which controls how much data is fed through the SOM at a time.

**Comment by ngreenwald** [comment]
In addition, we should add a batch-processing option to create_pixel_matrix, which right now relies on reading in the entire dataset at once

**Comment by alex-l-kong** [comment]
@ngreenwald to clarify, for `create_pixel_matrix`, you mean instead of passing the entire `img_xr` in, we just read in a few `fovs` at a time?

**Comment by ngreenwald** [comment]
Exactly. For larger datasets, it's not feasible to have the whole thing in memory at once.

**Comment by alex-l-kong** [comment]
@ngreenwald I wanted to clarify the batching process a bit.

1. In terms of normalization, we're actually not running it on a single `fov` at a time. `create_som_matrix.R` reads in the entire subsetted data from all the `fovs` before running 99.9% normalization. It then trains on the entire normalized subsetted dataset. Were you thinking of something else in terms of the batching?
2. In terms of preprocessing, I can now read in the image array just a few `fovs` at a time. However, due to the way Gaussian blurring and assigning the segmentation labels works, it's impossible to avoid needing to create pixel matrices per `fov`, which essentially negates the potential speedups we can get out of this. Still provides a massive space optimization, though.
3. In terms of training, were you thinking of training the `SOM` n times for n batches of `fovs` rather than at the same time? If so, unfortunately, I don't think that will be possible with `FlowSOM` the way it's implemented currently.
4. In terms of batching the R runner files in general, `feather` formats unfortunately aren't very well suited for reading in multiple files at a time. For clustering, I can still append n `fovs` to a matrix first before running the trained SOM (as opposed to immediately running it on data from a single `fov`), but I wouldn't be able to, say, read in 10 `fovs` into a matrix at a time.

**Comment by ngreenwald** [comment]
1. No, this is working as intended. We need to read this all in at once. 
2. If we don't get a speedup, that's okay. The main thing is switching to batch mode (even if the batch size is n=1) to allow cohorts of arbitrarily large size to be preprocessed
3. No, I hope I didn't suggest this to you. The point of subsetting is that we run on the entire (representative subsample) at once. There's no need to do multiple independent SOMs
4. Check and see if this is actually faster. If not, we can just leave it as n=1

**Comment by alex-l-kong** [comment]
@ngreenwald clustering on multiple `fovs` at a time takes significantly longer, I'll revert back to n=1.

**Comment by alex-l-kong** [comment]
This has been address by a previous PR.
